w3af/plugins/auth/detailed.py

```diff
@@ -58,7 +60,24 @@ def login(self):
 
         try:
             functor = getattr(self._uri_opener, self.method)
-            functor(self.auth_url, data)
+
+            redir_count = 0
+
+            while True:
+                if redir_count > 5:
+                    break
+                # actually file the auth HTTP reuqest
+                r = functor(self.auth_url, data)
+
+                # follow redirects if the feature is enabled
+                if self.follow_redirects and r.get_code() in GET_HEAD_CODES:
+                    self.auth_url.set_path(r.get_headers().iget('location')[0])
+                    om.out.debug('Auth redirected to new URL', self.auth_url)
+                    self.method = 'GET'
+                    data = ''
+                    redir_count += 1
+                else:
+                    break
```

Comment:
After looking at _web_spider_ I get your point and will try to think about how to implement this in more reusable way.